packages/http-server/src/http-server.ts

```diff
@@ -52,17 +61,37 @@ export interface HttpsOptions extends BaseHttpOptions, https.ServerOptions {
   protocol: 'https';
 }
 
+/**
+ * HTTP/2 HTTP server options
+ */
+export interface Http2Options extends BaseHttpOptions, http2.ServerOptions {
+  protocol: 'http2';
+}
+
+/**
+ * HTTP/2 HTTPS server options
+ */
+export interface Https2Options
+  extends BaseHttpOptions,
+    http2.SecureServerOptions {
+  protocol: 'https2';
```

Comment:
Nitpick: can we please use `http2s` as "http2" + "secure"? I find `https2` weird to read. 

Note that HTTP/2 spec puts the version number before additional flags too, e.g `h2c` for clear-text connections (not `hc2`). See e.g. https://stackoverflow.com/a/37373039/69868

```suggestion
  protocol: 'http2s';
```

Similarly the interface should be called `Http2SecureOptions` or perhaps `Http2sOptions` if you prefer.